tests/client_test.py

```diff
@@ -420,30 +395,35 @@ async def subscription_handler(msg):
         await nc.close()
 
     @async_test
-    async def test_subscribe_sync_call_soon(self):
+    async def test_subscribe_iterate(self):
         nc = NATS()
         msgs = []
+        fut = asyncio.Future()
 
-        def subscription_handler(msg):
-            msgs.append(msg)
+        async def iterator_func(sub):
+            async for msg in sub.messages:
+                msgs.append(msg)
+            fut.set_result(None)
 
         await nc.connect()
-        sid = await nc.subscribe("tests.>", cb=subscription_handler)
+        sub = await nc.subscribe('tests.>', iterate=True)
```

Comment:
How about if `cb` being `None` would mean that the subscription has to be used in the iterator style? That would feel slightly more similar to how the [nats.rs/Rust client does it](https://github.com/nats-io/nats.rs/blob/b6e04c4768a0580634dfb03c4ca252b2ebf245f1/examples/nats-box/main.rs#L73), and also would not need to have `subscribe_sync` like in the Go client.  Wdyt?

```python
sub = await nc.subscribe('tests.>')
...
async for msg in sub.messages:
...
```

Reply:
Yeah I'm fine with that. I need to edit the docs still, I'll add something that explicitly says that so consumers are aware.